cli/cli-api/src/main/java/com/quorum/tessera/cli/parsers/ServerURIOutputMixin.java

```diff
@@ -0,0 +1,111 @@
+package com.quorum.tessera.cli.parsers;
+
+import com.quorum.tessera.config.AppType;
+import com.quorum.tessera.config.Config;
+import com.quorum.tessera.server.TesseraServer;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+public class ServerURIOutputMixin {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ServerURIOutputMixin.class);
+
+  @CommandLine.Option(
+      names = {"--XoutputServerURIs"},
+      description = "Output the server URI(s) to a specified path",
+      hidden = true)
+  private Path outputServerURIPath = null;
+
+  public void updateConfig(final Config config) {
+    updateConfig(this.outputServerURIPath, config);
+  }
+
+  public void updateConfig(final Path outputServerURIPath, final Config config) {
+    if (outputServerURIPath != null) {
+      config.setOutputServerURIPath(outputServerURIPath);
+    }
+  }
+
+  public static void writeServerURIsToFile(
+      final Path outputServerURIPath, final List<TesseraServer> servers) {
+    try {
+      final TesseraServer q2tServer =
```

Comment:
Minor style issue: this code is quite repetitive. Couldn't we have a single loop of `servers`, call `writeURIFile` on every single element and just do nothing if `fileNameFromAppType` returns `null`?

I am assuming we do not need the `findFirst()` on the stream to eliminate duplicate instances of the same service.

Reply:
Good suggestion.